packages/canvas-panel-core/src/components/ThumbnailList/ThumbnailList.js

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withBemClass } from '@canvas-panel/core';
+//import './ThumbnailList.scss';
+
+const thumbnailGetSize = (thumbnail, pWidth, pHeight) => {
+  const thumb = thumbnail.__jsonld;
+  if (
+    (pWidth || pHeight) &&
+    thumb.hasOwnProperty('service') &&
+    thumb.service.hasOwnProperty('sizes')
+  ) {
+    let closestSizeIndex = -1;
+    let minDistanceX = Number.MAX_SAFE_INTEGER;
+    let minDistanceY = Number.MAX_SAFE_INTEGER;
+    thumb.service.sizes.forEach((size, index) => {
+      if (pWidth) {
+        const xDistance = Math.abs(size.width - pWidth);
+        if (minDistanceX >= xDistance) {
+          closestSizeIndex = index;
+          minDistanceX = xDistance;
+        }
+      }
+      if (pHeight) {
+        const yDistance = Math.abs(size.height - pHeight);
+        if (minDistanceY >= yDistance) {
+          closestSizeIndex = index;
+          minDistanceY = yDistance;
+        }
+      }
+    });
+    let thumbUrlParts = (thumb.id || thumb['@id']).split('/');
+    if (closestSizeIndex !== -1) {
+      const size = thumb.service.sizes[closestSizeIndex];
+      thumbUrlParts[thumbUrlParts.length - 3] = [size.width, size.height].join(
+        ','
+      );
+    }
+    return thumbUrlParts.join('/');
+  } else {
+    return thumb.id || thumb['@id'];
+  }
+};
+
+class ThumbnailList extends React.Component {
+  thumbnailCache = {};
+
+  componentDidUpdate(/*prevProps, prevState*/) {
+    if (this.selectedThumbnail) {
+      const list = this.selectedThumbnail.parentNode.parentNode;
+      const rect = this.selectedThumbnail.getBoundingClientRect();
+      if (rect.x < 0) {
+        list.scrollLeft = 0;
+      } else if (list.offsetWidth - rect.width < rect.x) {
+        list.scrollLeft =
+          this.selectedThumbnail.offsetLeft - (list.offsetWidth - rect.width);
```

Comment:
I can see that this ensures the current thumbnail is in view (by moving it to the left), it would be nice to be able to configure this so that the selected thumbnail is in the middle of the `parentNode`

Reply:
https://github.com/digirati-co-uk/canvas-panel/pull/370/commits/745d2c2bc4bca9e8041063bd604b5acc8e17c5c4